statsmodels/graphics/gofplots.py

```diff
@@ -178,20 +187,25 @@ def sorted_data(self):
         return sorted_data
 
     @cache_readonly
-    def sample_quantiles(self):
-        if self.fit and self.loc != 0 and self.scale != 1:
-            return (self.sorted_data-self.loc)/self.scale
+    def scaled_data(self):
+        if self.loc == 0 and self.scale == 1:
+            return (self.sorted_data-np.mean(self.data))/np.std(self.data)
         else:
-            return self.sorted_data
+            return self.sample_quantiles
+
+    @cache_readonly
+    def sample_quantiles(self):
+        return self.sorted_data #(self.sorted_data-self.loc)/self.scale
 
     @cache_readonly
     def sample_percentiles(self):
-        quantiles = \
-            (self.sorted_data - self.fit_params[-2])/self.fit_params[-1]
-        return self.dist.cdf(quantiles)
+        if self.fit or self._userdist_is_frozen:
+            return self.dist.cdf(self.sorted_data)
+        else:
+            return self.dist.cdf(self.scaled_data)
```

Comment:
This automatic switching between sorted and scaled data looks "strange"
If user has fully specified (distargs, loc and scale) but not frozen distribution, then this would scale data again, while it should just use the unscaled data.